python_lambda_logging/lambda_logging.py

```
@@ -1,40 +1,45 @@
+"""Lambda logging decorator to standarize logging."""
 import logging
-import functools
 
 
 def setup_lambda_logger():
-    """
-    A utility function for configuring python logging for use in lambda functions using the format:
+    r"""
+    A utility function for configuring python logging for use in lambda functions using the format.
+
     %(levelname)s RequestId: %(aws_request_id)s\t%(message)s\n
     """
     logger = logging.getLogger()
-    for h in logger.handlers:
-        FORMAT = '%(levelname)s RequestId: %(aws_request_id)s\t%(message)s\n'
-        h.setFormatter(logging.Formatter(FORMAT))
-    
+    for handler in logger.handlers:
+        logformat = '%(levelname)s RequestId: %(aws_request_id)s\t%(message)s\n'
+        handler.setFormatter(logging.Formatter(logformat))
+
     logger.setLevel(logging.INFO)
-    
+
     return logger
-    
+
 
 def logged_handler(logger):
     """
-    A decorator that wraps a lambda_handler and logs the function name, event, return value and 
-    any exception if one is raised
+    A decorator that wraps a lambda_handler.
+
+    This logs the function name, event, return value and any exception if one is raised.
     """
     def decorator(function):
         def wrapper(*args, **kwargs):
             event = args[0]
             context = args[1]
-            logger.info("Function: " + context['invoked_function_arn'] + " - " + context['function_version'])
-            logger.info("Event: " + str(event))
-
+            try:
+                logger.info("Function: %s - %s", context['invoked_function_arn'], context['function_version'])
+            except TypeError as expt:
+                logger.error("Invalid Context: %s %s", expt, context)
+                raise expt
```

do you think this should raise an error? @robgodfrey-ft

---

Probably not - should log what information it can from the context safely i.e. if it's available). Instead of context['invoked_function_arn'] could we use context.get('invoked_function_arn', 'invoked_function_arn not supplied') i.e. provide a default for when not present